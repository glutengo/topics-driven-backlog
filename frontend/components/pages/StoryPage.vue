<template>
    <section id="story-page">
        <template v-if="story">
            <h1>
                {{ story.title }}
                <small class="text-muted">{{ story.identifier }}</small>
            </h1>

            <p v-if="story.description">
                {{ story.description }}
            </p>
        </template>
        <NotFound v-else/>
    </section>
</template>

<script>
import NotFound from '@/components/elements/NotFound';

export default {
    components: { NotFound },
    data() {
        return {};
    },
    computed: {
        /**
         * Story to display
         * @returns {object}
         */
        story() {
            return this.$store.getters['stories/byIdentifier'](
                this.$route.params.identifier
            );
        },
    },
};
</script>