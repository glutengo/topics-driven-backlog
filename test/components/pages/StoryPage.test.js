import { mount, createLocalVue } from 'vue-test-utils';
import mockRouter from 'mock-vue-router';
import Vuex from 'vuex';
import StoryPage from '@/components/pages/StoryPage';

const localVue = createLocalVue();
localVue.use(Vuex);

const { $router, $route } = mockRouter(['/stories/:identifier'], '/stories/1');

describe('StoryPage.test.js', () => {
    let cmp, store, getters;

    beforeEach(() => {
        getters = {
            byIdentifier: () => id =>
                [
                    { id: 1, title: 'Test story' },
                    { id: 2, title: 'Test story 2' },
                ][id - 1],
        };

        store = new Vuex.Store({
            modules: {
                stories: {
                    namespaced: true,
                    getters,
                },
            },
        });

        cmp = mount(StoryPage, {
            localVue,
            store,
            mocks: {
                $router,
                $route,
            },
        });
    });

    it('returns correct story from route param', () => {
        expect(cmp.vm.story.title).toBe('Test story');
    });

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot();
    });
});
