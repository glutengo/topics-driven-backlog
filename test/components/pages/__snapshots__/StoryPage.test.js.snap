// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryPage.test.js has the expected html structure 1`] = `
<section
  id="story-page"
>
  <h1>
    
            Test story
            
    <small
      class="text-muted"
    >
      
    </small>
  </h1>
   
  <!---->
</section>
`;
